menpo/model/pca.py

```diff
@@ -263,8 +373,8 @@ def noise_variance(self):
         noise_variance : `float`
             The mean variance of the inactive components.
         """
+        noise_variance = 0.0
         if self.n_active_components == self.n_components:
-            noise_variance = 0.0
             if self._trimmed_eigenvalues.size is not 0:
                 noise_variance += self._trimmed_eigenvalues.mean()
```

Comment:
better as
```py
if (self.n_active_components == self.n_components and 
    self._trimmed_eigenvalues.size != 0):
```
right?

To be clear:

1. nested if's are redunant and more pressingly 
2. the `is` equality check wiht 0 is dangerous